### PR TITLE
Core/Waypoints: Refactor to split data into path and node related inf…

### PR DESCRIPTION
**Pull request by mdX7**
…ormation in db

**Changes proposed:**
-  refactor waypoints to have one table for path related stuff and one table for node related stuff only

**Tests performed:**
tested ingame

### CODE REVIEW

File: src/server/game/Movement/Waypoints/WaypointDefines.h [hunk 1]
```diff
@@ -34,36 +50,41 @@ enum WaypointMoveType
...
 
-    uint32 id;
-    float x, y, z;
-    Optional<float> orientation;
-    uint32 delay;
-    uint32 moveType;
+    uint32 Id;
+    float X;
+    float Y;
+    float Z;;
```
Comment (Nyr97) [inline src/server/game/Movement/Waypoints/WaypointDefines.h:68]: Double ;;

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 2]
```diff
@@ -2066,17 +2066,17 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
...
                             WaypointPath const* path = sWaypointMgr->GetPath(pathId);
-                            if (!path || path->nodes.empty())
+                            if (!path || path->Nodes.empty())
                                 continue;
 
-                            for (WaypointNode const& waypoint : path->nodes)
+                            for (WaypointNode const& waypoint : path->Nodes)
                             {
-                                float distamceToThisNode = creature->GetDistance(waypoint.x, waypoint.y, waypoint.z);
+                                float distamceToThisNode = creature->GetDistance(waypoint.X, waypoint.Y, waypoint.Z);
```
Comment (Nyr97) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:2074]: distamce to distance, so we fix this spelling uhu

Comment (Ovahlord) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: u sure you didn't wanna use GetExactDist here?

Comment (Nyr97) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: well that means changing the getter which originally was GetDistance

**Pull request merged**